### ISSUE TITLE
Improve dashboard `/searches` filters

### ISSUE THREAD

**Issue by githubfan21** [serpapi/public-roadmap#125]
![image](https://user-images.githubusercontent.com/307597/165848184-640f30a6-843b-440b-a960-4cd8333a3188.png)

Please add the following filters to "Your Searches" dashboard page as it is very hard to find specific records.

Filter examples:
- engine
- query text box search
- device
- date/date range
- location
- google domain
- language

**Comment by aliayar** [comment]
Hi,

Please check our guidelines for [reporting an issue](https://github.com/serpapi/public-roadmap#report-an-issue) and [asking for a new feature](https://github.com/serpapi/public-roadmap#ask-for-new-apis-or-features).

**Comment by githubfan21** [comment]
why is the issue closed? It does meet your guidelines?

**Comment by hartator** [comment]
Added a screenshot and clarified some wordings @githubfan21. 👍

We might not be able to deliver on this as we don't have a straightforward way to make lookup fast without restructuring our database.

**Comment by schaferyan** [comment]
Another customer requested this:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/view/99173/conversation/102910000053115?view=List)

**Comment by schaferyan** [comment]
This feature could also be helpful for the Customer Success team, as it could make it easier to find past examples of an issue

**Comment by kagermanov27** [comment]
Also mentioned in https://github.com/serpapi/SerpApi/issues/1250

**Comment by alexbarron** [comment]
Received another request for this.

[Front](https://app.frontapp.com/open/cnv_1bgdg1p8?key=B9mq42nnpTSadwqtsA-s0CfnMP8_Sxcy)

**Comment by martin-serpapi** [comment]
Another customer requested this:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000071046?view=List)

**Comment by marm123** [comment]
A high-volume customer requested this:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000071075?view=List)

**Comment by marm123** [comment]
An Enterprise Customer requested this:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000072085?view=List)

**Comment by sanj-directbooker** [comment]
Another request from an enterprise customer

**Comment by marm123** [comment]
Another Enterprise Customer requested this:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000078617)

**Comment by alexbarron** [comment]
Another request for date filtering:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/215468972916205?view=List)

**Comment by marm123** [comment]
Another high-volume customer requested this:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/215468990973858?view=List)

**Comment by alexbarron** [comment]
Another request, specifically for searching:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/215469005261014?view=List)

**Comment by hilmanski** [comment]
The customer requested this.

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/215469577222755?view=List)

**Comment by hilmanski** [comment]
A customer is interested in filtering by the "q" parameter.

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/215472388509326?view=List)

**Comment by oceansize** [comment]
A customer requested this:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/215473239003170?view=List)

**Comment by NateSkiles** [comment]
A customer requested this:

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/215473252810389)

**Comment by oceansize** [comment]
Another request:

[Front](https://app.frontapp.com/open/msg_2zm9zw7w?key=KcX3Eo9QAlKk3SASQFb_-VDyk7x9urd9)

**Comment by trusche** [comment]
We've now added a `Created At` column and the ability to filter searches by date range:

<img width="2286" height="870" alt="Image" src="https://github.com/user-attachments/assets/99819b57-8972-4f09-a3f6-5bb02cc71466" />

We're still looking into filtering by other parameters.

**Comment by NateSkiles** [comment]
@trusche, just to confirm—we are keeping this issue open until search parameter filters are added, correct?

**Comment by trusche** [comment]
> we are keeping this issue open until search parameter filters are added, correct?

@NateSkiles Yes, or until we've determined that we won't add them because of performance trade-offs.

**Comment by sonika-serpapi** [comment]
[Tracking Sheet](https://docs.google.com/spreadsheets/d/1TjK3sFVOmyp1x2DP4AuvAnKUvKaX83CLr1JWcX36Dp0/edit?gid=0#gid=0) for more enquiries

**Comment by trusche** [comment]
@NateSkiles @schaferyan @oceansize We added the date filter a while ago. Any other fields would require additional indexing and come with a performance hit that we probably do not want to incur. Ultimately, I don't think our searches page was ever meant to be a fully searchable archive, given that searches expire anyway after 30 days. I think we can at least de-prioritize this, or even freeze it. WDYT?

**Comment by roi-serpapi** [comment]
> [@NateSkiles](https://github.com/NateSkiles) [@schaferyan](https://github.com/schaferyan) [@oceansize](https://github.com/oceansize) We added the date filter a while ago. Any other fields would require additional indexing and come with a performance hit that we probably do not want to incur. Ultimately, I don't think our searches page was ever meant to be a fully searchable archive, given that searches expire anyway after 30 days. I think we can at least de-prioritize this, or even freeze it. WDYT?

@trusche I think if we are able to close #65, it would definitely support freezing this.
